### ISSUE TITLE
Tracker URL remover: "IndexError: pop index out of range"

### ISSUE THREAD

**Issue by x-0n** [searx/searx#1734]
**Issue:**
Certain searches (I believe it's searches that return a number of results larger than some magic number) fail with a _"search error"_ in the web UI and the following in _/var/log/uwsgi/uwsgi.log:_

```
ERROR:searx.webapp:search error
Traceback (most recent call last):
  File "/usr/local/searx/searx/webapp.py", line 514, in index
    result_container = search.search()
  File "/usr/local/searx/searx/search.py", line 545, in search
    plugins.call(self.ordered_plugin_list, 'on_result', self.request, self, result)
  File "/usr/local/searx/searx/plugins/__init__.py", line 73, in call
    ret = getattr(plugin, plugin_type)(request, *args, **kwargs)
  File "/usr/local/searx/searx/plugins/tracker_url_remover.py", line 46, in on_result
    parsed_query.pop(i)
IndexError: pop index out of range
```

The issue disappears when I disable _Tracker URL remover_ in search preferences.

**Steps to repeat:**
- enable _Tracker URL remover_
- enter a query that will return a large number of results. Try _"assange", time:any_

**Comment by MarcAbonce** [comment]
Are you using the latest commit from master?
Because I supposedly fixed that error a couple of days ago in #1729, but maybe I missed some other edge case.

Based on the uwsgi log you posted though, I'm pretty sure your instance is based on an older commit because the line 46 in your `tracker_url_remover.py` was already changed in that PR.

If your instance is based on a Docker image, then I don't know how often they are updated with master.

**Comment by x-0n** [comment]
@MarcAbonce I'm sorry, should have mentioned. I'm using the docker image at searx/searx:latest.

The good news is that after updating my docker image the error has gone away. So your fix is apparently working :)

The bad news is that I don't find that PR in the build history here, so no idea if it was actually your fix that fixed this or something else: https://travis-ci.org/searx/searx/builds

Before updating my docker image today, I should have been at https://travis-ci.org/searx/searx/builds/603251099

**Comment by dalf** [comment]
@x-0n : The last docker image version is ```0.15.0-186-42d5e2c0```, it matches the commit 42d5e2c02cd4715a0e09411efbb249ef5d8defed

The [travis build 603251099](https://travis-ci.org/searx/searx/builds/603251099) is about the commit 12f42d1.
Line 479, you can see the docker version : ```searx/searx:0.15.0-177-12f42d15```.

Looking at https://github.com/searx/searx/commits/master, the PR merged the commit 147ad504c340727b77b1e23568a29c1daef0df8b which is after the commit 12f42d15.

Note, that when you use the docker image, you can see the version and commit at the bottom of the web pages, example here: https://a.searx.space

**Comment by x-0n** [comment]
@dalf : thanks for clarifying :)

**Comment by asciimoo** [comment]
fixed in dede46a2
